modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/FieldSelectModalContent.tsx

```diff
@@ -298,36 +312,51 @@ const FieldSelectModalContent = ({
 									query,
 								}: IFieldTreeItem) => (
 									<TreeView.Item>
-										<TreeView.ItemStack
-											disabled={
-												initialChildren &&
-												!!initialChildren.length
-													? true
-													: false
-											}
-											expanderDisabled={false}
-										>
-											<ClayCheckbox checked>
-												<span className="font-weight-normal pl-1 text-3">
+										{selectionMode === 'single' ? (
+											<TreeView.ItemStack
+												disabled={
+													!!initialChildren?.length
+												}
+												expanderDisabled={false}
+											>
+												<ClayCheckbox checked />
+
+												<Highlight
+													query={query}
+													text={label}
+												/>
+											</TreeView.ItemStack>
```

Comment:
For single selection, we need checkbox and label to be siblings. For multiple selection, we need label to be nested in checkbox component. Sibling structure in the expectation of the Clay component, nested structure is our workaround to have interactive labels in multiple selection.